csi.proto

```diff
@@ -1296,6 +1345,55 @@ message VolumeUsage {
   // Units by which values are measured. This field is REQUIRED.
   Unit unit = 4;
 }
+
+message VolumeHealth {
+  option (alpha_message) = true;
+
+  enum Condition {
+    // Volume health condition is unknown.
+    UNKNOWN = 0;
+    // Volume is accessible and no problems are detected. 
+    HEALTHY = 1;
+    // Volume is not accessible currently, but there's no reason to
+    // expect data loss. Examples: power outage, network outage,
+    // maintenance.
+    TEMPORARY_INACCESSIBLE = 10;
+    // Volume is accessible currently, but is degraded. Data loss
+    // is unlikely as the storage system may be rebuilding to fix the
+    // problem automatically. For example, a replica is lost on
+    // the storage system but a spare is configured, and the storage
+    // system is resyncing/reconstructing.
+    TEMPORARY_DEGRADED = 20;
+    // Volume is accessible currently, but a problem is detected that
+    // could lead to eventual data loss. Examples: recoverable hardware
+    // failure (i.e. the disk needs to be manually replaced by a human),
+    // cooling failure, equipment significantly past end of life.
+    FAILURE_LIKELY = 30;
+    // Volume is not accessible currently, and there is reason to
+    // believe the outage may be permanent. Examples: unrecoverable
+    // hardware failure, natural disasters, damage to equipment.
+    FATAL = 40;
+  }
+
+  message VolumeHealthError {
+    // The code describing the health condition of the volume.
+    // This is an opaque field to CO.
+    // This field is OPTIONAL.
+    string code = 1;
```

Comment:
Why have a separate code and message if this is intended to be opaque? This seems like a hole in the API that will permit a high level consumer to implement logic specific to a storage vendor. We should discourage that via CSI (for those who want that they should have the storage vendor expose custom APIs as needed).

Reply:
Sure, I'll remove code.

Reply:
Done